### PR TITLE
`level` output for the TextureBlock / switch for disabling level multiplication

### PR DESCRIPTION
**Pull request by rassie**
This PR is a playground for the discussion in https://forum.babylonjs.com/t/nme-make-texture-level-work-for-normal-maps-as-it-does-in-other-materials/19443/14. This code currently does not work (at all!), but might work someday.

### CODE REVIEW

**Comment by sebavan** [comment]
LGTM aside the // FIXME block which needs to be removed. @Popov72  will provide a more in depth analysis :-)

**Comment by rassie** [comment]
> LGTM aside the // FIXME block which needs to be removed.

Yeah, I couldn't find the reason for that. Would look into it again later today, but would also appreciate any help with that. Maybe there is something obvious that I've overlooked.

**Comment by rassie** [comment]
@sebavan @Popov72 have found and fixed the FIXME problem. This patch should be complete now.

**Comment by sebavan** [comment]
@rassie, you have one lint issue:
image

And you should credit yourself in the dist/preview release/whats new.md file

So that the build will be all green and happy.

@Popov72 will do the thorough review soon.

**Comment by rassie** [comment]
Done!

**Review by sebavan — APPROVED** [review]

File: src/Materials/Node/Blocks/Dual/textureBlock.ts [hunk 1]
```diff
@@ -73,6 +73,11 @@ export class TextureBlock extends NodeMaterialBlock {
      */
     public convertToLinearSpace = false;
 
+    /**
+     * Gets or sets a boolean indicating if content needs to be converted to linear space
+     */
+    public disableLevelMultiplication = false;
```
Comment (Popov72) [inline src/Materials/Node/Blocks/Dual/textureBlock.ts:79]: nit: The doc is wrong

File: src/Shaders/ShadersInclude/bumpFragment.fx [hunk 2]
```diff
@@ -14,7 +14,7 @@
 	#elif defined(BUMP)
 		// flip the uv for the backface
 		vec2 TBNUV = gl_FrontFacing ? vBumpUV : -vBumpUV;
-		mat3 TBN = cotangent_frame(normalW * normalScale, vPositionW, TBNUV);
+		mat3 TBN = cotangent_frame(normalW * normalScale, vPositionW, TBNUV,  vTangentSpaceParams);
```
Comment (Popov72) [inline src/Shaders/ShadersInclude/bumpFragment.fx:17]: nit: two spaces instead of one

**Review by Popov72 — COMMENTED** [review]
LGTM: Great job!

One small thing is that 3 functions have been removed from `bumpFragmentFunctions.fx` and that could be a breaking change if someone was using them...

However, I don't think this code was supposed to be called by external users and having `3 x perturbNormal + 2 x cotangent_frame` functions reduced to `1 x perturbNormal + 1 x cotangent_frame` sounds good to me, so let's keep it!

**Comment by rassie** [comment]
> LGTM: Great job!
> 
> One small thing is that 3 functions have been removed from `bumpFragmentFunctions.fx` and that could be a breaking change if someone was using them...
> 
> However, I don't think this code was supposed to be called by external users and having `3 x perturbNormal + 2 x cotangent_frame` functions reduced to `1 x perturbNormal + 1 x cotangent_frame` sounds good to me, so let's keep it!

It's a good thing there is a 5.0 release coming then ;) I'll fix the nits in a couple of hours when I'm not on mobile! Thanks!

**Comment by rassie** [comment]
Nits are done, thanks for the feedback!

**Pull request merged**